databricks/koalas/series.py

```diff
@@ -3224,6 +3224,100 @@ def copy(self) -> 'Series':
         """
         return _col(DataFrame(self._internal.copy()))
 
+    def truncate(self, before=None, after=None, copy=True):
+        """
+        Truncates a sorted Series before and/or after some particular index value.
+        Series should have sorted index.
+
+        .. note:: the current implementation of truncate uses is_monotonic_increasing internally
+            This leads to move all data into single partition in single machine and could cause
+            serious performance degradation. Avoid this method against very large dataset.
+
+        Parameters
+        ----------
+        before : string, int
+            Truncate all rows before this index value
+        after : string, int
+            Truncate all rows after this index value
+        copy : boolean, default is True,
+            return a copy of the truncated section
+
+        Returns
+        -------
+        truncated : Series
+
+        Examples
+        --------
+
+
+        A Series has index that sorted integers.
+
+        >>> s = ks.Series([10, 20, 30, 40, 50, 60, 70],
+        ...               index=[1, 2, 3, 4, 5, 6, 7])
+        >>> s
+        1    10
+        2    20
+        3    30
+        4    40
+        5    50
+        6    60
+        7    70
+        Name: 0, dtype: int64
+
+        >>> s.truncate(2, 5)
+        2    20
+        3    30
+        4    40
+        5    50
+        Name: 0, dtype: int64
+
+        A Series has index that sorted strings.
+
+        >>> s = ks.Series([10, 20, 30, 40, 50, 60, 70],
+        ...               index=['a', 'b', 'c', 'd', 'e', 'f', 'g'])
+        >>> s
+        a    10
+        b    20
+        c    30
+        d    40
+        e    50
+        f    60
+        g    70
+        Name: 0, dtype: int64
+
+        >>> s.truncate('b', 'e')
+        b    20
+        c    30
+        d    40
+        e    50
+        Name: 0, dtype: int64
+        """
+        indexes = self.index
+        indexes_increasing = indexes.is_monotonic_increasing
+        if not any([indexes_increasing, indexes.is_monotonic_decreasing]):
```

Comment:
@itholic can you just use this condition?

```python
not indexes.is_monotonic_increasing or not indexes.is_monotonic_decreasing
```

It will allow a short circuit and we can avoid to call both `is_monotonic_increasing` and `is_monotonic_decreasing`.

Reply:
@HyukjinKwon thanks for commenting!

but maybe this will raise `ValueError` even when `indexes.is_monotonic_increasing` is `False` and `indexes.is_monotonic_decreasing` is `True` ?? 

in this case, we should pass them without raises Error.

Reply:
Oh, then maybe `not (indexes.is_monotonic_increasing or indexes.is_monotonic_decreasing)`

Reply:
I see. can we simply like this: `not indexes.is_monotonic_increasing and not indexes.is_monotonic_decreasing`?

Reply:
okay i agree with that is the more readable. thanks for reviewing!!